packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -0,0 +1,358 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'box.dart';
+import 'object.dart';
+
+enum WrapAlignment {
+  start,
+  center,
+  end,
+  // TODO(ianh): justify
+}
+
+class WrapParentData extends ContainerBoxParentDataMixin<RenderBox> { }
+
+class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, WrapParentData>,
+                                        RenderBoxContainerDefaultsMixin<RenderBox, WrapParentData> {
+  RenderWrap({
+    List<RenderBox> children,
+    Axis direction: Axis.horizontal,
+    WrapAlignment mainAxisAlignment: WrapAlignment.center,
+    WrapAlignment crossAxisAlignment: WrapAlignment.center,
```

Comment:
So, there are actually two kinds of alignment in the cross axis.  The first is the alignment within a run.  For example, if we're horizontal, then all the boxes in a row a vertically centered w.r.t each other.  The second is the alignment of the runs themselves within the overall box.  For example, you might want all the rows as a group vertically centered within the box.

A value like justify makes sense for the second type of alignment, but I'm not sure it makes sense for the first.  That means you might want to use a different enum for `crossAxisAlignment` (and maybe pick a name that makes it easy to add the other kind of cross axis alignment in the future).